onboarding/main.go

```diff
@@ -57,90 +56,87 @@ func main() {
 	// Instead, a safer approach is to delete both secrets and then recreate them simultaneously
 	// to ensure consistency and accuracy of all secrets. By this way it will be easier to diagnose the
 	// issues if one or two secrets do not exist instead of trying to understand if they match
-	err = clientset.CoreV1().
-		Secrets(operatorNamespace).
-		Delete(ctx, onboardingValidationPrivateKeySecretName, metav1.DeleteOptions{})
+	privateSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      onboardingValidationPrivateKeySecretName,
+			Namespace: operatorNamespace,
+		},
+	}
+	err = cl.Delete(ctx, privateSecret)
```

Comment:
Just a suggestion for shorter code with less nesting:
```suggestion
	privateSecret := &corev1.Secret{}
	privateSecret.Name = onboardingValidationPrivateKeySecretName
	privateSecret.Namespace =  operatorNamespace
	err = cl.Delete(ctx, privateSecret)
```